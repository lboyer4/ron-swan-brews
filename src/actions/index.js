export const setBreweries = (breweries) => ({
	type: 'SET_BREWERIES',
	breweries
})	