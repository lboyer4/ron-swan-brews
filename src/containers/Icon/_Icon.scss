nav {
	display: flex;
	padding: 10px;
}

.title {
	margin-left: 5px;
}

.icon-holder {
	color: #fff;
}

.ronswan-quote {
	padding: 10px;
}