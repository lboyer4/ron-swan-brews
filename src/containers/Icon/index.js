import React, { Component } from 'react';
import './_Icon.scss';
import ronswanIcon from '../../utils/images/ronswanIcon.png'

class Icon extends Component {
	constructor() {
		super();
		this.state = {

		}
	}

	render() {
		return (
			<div className="icon-holder">
				<nav>
					<img src={ronswanIcon} height="100px" alt="icon with an outline of ron swanson's face and hair" />
					<h1 className="title">RonSwanBreweries</h1>
				</nav>
				<p className="ronswan-quote">quote will eventually go here and wont that be nice</p>
			</div>
			)
	}
}

export default Icon;