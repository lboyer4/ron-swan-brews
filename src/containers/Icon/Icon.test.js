import React from 'react';
import { shallow } from 'enzyme';
import { Icon, mapStateToProps, mapDispatchToProps } from '../Icon';
import { setSearch } from '../../actions';

describe('Icon', () => {
	let wrapper;
	let mockSetSearch;
	let mockQuotes;

	beforeEach(() => {
		mockSetSearch = jest.fn()
		mockQuotes = ['one', 'two']
		wrapper = shallow(
			<Icon 
				setSearch = {mockSetSearch}
				quotes = {mockQuotes}
			/>
		)
	});

	it('should match the snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});
	describe('clearSearch', () => {
		it('should clear search', () => {
			wrapper = shallow(
				<Icon
					search= ''
					setSearch={jest.fn()}
				/>)
			wrapper.instance().clearSearch()
			expect(wrapper.instance().props.setSearch).toHaveBeenCalled()
		});
	});

	describe('mapStateToProps', () => {
		it('should return a props array of quotes', () => {
			const mockState = {
				breweries: ['one', 'two'],
				quotes: ['to be', 'not to be'],
				search: 'Denver'
			}
			const expected = {
				quotes: ['to be', 'not to be'],
			}
			const mappedProps = mapStateToProps(mockState)

			expect(mappedProps).toEqual(expected);
		});
	});

	describe('mapDispatchToProps', () => {
		it('should dispatch set Search', () => {
			const mockDispatch = jest.fn();
			const actionToDispatch = setSearch('');
			const mappedProps = mapDispatchToProps(mockDispatch);

			mappedProps.setSearch('');
			expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
		});
	});
});