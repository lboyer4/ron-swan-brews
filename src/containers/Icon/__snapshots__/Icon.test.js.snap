// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should match the snapshot 1`] = `
<div
  className="icon-holder"
>
  <nav>
    <Link
      onClick={[Function]}
      to="/"
    >
      <img
        alt="icon with an outline of ron swanson's face and hair"
        height="100px"
        src="ronswanIcon.png"
      />
    </Link>
    <div
      className="title"
    >
      <h1>
        RonSwanBreweries
      </h1>
      <h4
        className="ronswan-quote"
      >
        "Clear alcohols are for rich women on diets."
      </h4>
      <p>
         -Ron Swanson
      </p>
    </div>
  </nav>
</div>
`;
